src/DataAccess/DoctrineDonationRepository.php

```diff
@@ -207,15 +207,11 @@ public function getDonationById( int $id ) {
 	 * @return DoctrineDonation|null
 	 * @throws ORMException
 	 */
-	private function getDoctrineDonationById( int $id ) {
-		/*
-		 * FIXME: only select donations that have
-			 * 'isPublic' => true,
-			 * 'dtDel' => null
-			 * correct status
-		 */
-
-		return $this->entityManager->find( DoctrineDonation::class, $id );
+	public function getDoctrineDonationById( int $id ) {
+		return $this->entityManager->getRepository( DoctrineDonation::class )->findOneBy( [
+			'id' => $id,
+			'dtDel' => null
```

Comment:
Ye Olde Application™ also sets this timestamp, when the dataset is updated with an inactive status (**D**eleted, **P**ending, in**X**omplete, promi**Z**ed bank transfer). I think only the first one really makes sense, the other three don't really relate to a deletion/cancellation.

I think both the deletion in the management/backend application and the cancellation by the user should trigger this timestamp to be set.